packages/react-live/src/utils/transpile/index.ts

```diff
@@ -3,22 +3,40 @@ import transform from "./transform";
 import errorBoundary from "./errorBoundary";
 import evalCode from "./evalCode";
 import compose from "./compose";
-import { Transform } from "sucrase";
+import { Transform, Options as SucraseOptions } from "sucrase";
 
 const jsxConst = 'const _jsxFileName = "";';
 const trimCode = (code: string) => code.trim().replace(/;$/, "");
 const spliceJsxConst = (code: string) => code.replace(jsxConst, "").trim();
 const addJsxConst = (code: string) => jsxConst + code;
 const wrapReturn = (code: string) => `return (${code})`;
 
-type GenerateOptions = {
+export type GenerateOptions = {
   code: string;
   scope?: Record<string, unknown>;
   enableTypeScript: boolean;
+  jsxRuntime?: SucraseOptions["jsxRuntime"];
 };
 
+function createRequire() {
+  return (val: string) => {
+    if (val === "react/jsx-runtime") {
+      return require("react/jsx-runtime");
+    } else if (val === "react/jsx-dev-runtime") {
+      return require("react/jsx-dev-runtime");
+    } else {
+      return require(val);
+    }
+  };
+}
```

Comment:
I'm not super fan of this part, any idea how to deal with the new `require("react/jsx-runtime")` that Sucrase adds?

The transformed code looks like this now:

```tsx
const _jsxFileName = "";
("use strict");
var _jsxdevruntime = require("react/jsx-dev-runtime");
return _jsxdevruntime.jsxDEV.call(
  void 0,
  "div",
  { children: "Hello World!" },
  void 0,
  false,
  { fileName: _jsxFileName, lineNumber: 1 },
  this,
);

```


Maybe it would be better to use Sucrase to remove the require call directly, and inject a `_jsxdevruntime` variable in scope directly?

Can injecting a `require()` function in scope be a problem?